db/system_keyspace.cc

```diff
@@ -2526,6 +2535,11 @@ future<service::topology> system_keyspace::load_topology_state() {
             rebuild_option = row.get_as<sstring>("rebuild_option");
         }
 
+        std::unordered_set<raft::server_id> ignored_ids;
```

Comment:
> In system_keyspace::load_topology_state, we don't need to emplace
ignored_ids into ret.req_param if a row contains topology_request.
This parameter will only be used after deleting topology_request.

But I see that you emplace it unconditionally nevertheless. Is that paragraph of the commit message important in any way?

Reply:
The code emplaces an empty set in this case, but why the complication?

Reply:
> The code emplaces an empty set in this case, but why the complication?

Right, it's not worth doing it. I changed it to emplace `ignored_ids` for the replace operation in both cased.

The paragraph still applies to the removenode operation, but I think it brings more confusion than help, so I removed it.